ksql-parser/src/main/java/io/confluent/ksql/parser/tree/Node.java

```diff
@@ -24,6 +24,8 @@
 
   private final Optional<NodeLocation> location;
 
+  private Optional<Node> parent;
```

Comment:
Is this intended to be tri-state i.e. null, present, empty?  Or, as per `location` is `Optional` being used here to make it explicit that the return value of `getParent` may be "no node"?

If it's not meant to be tri-state, then consider either wrapping in Optional on access, e.g.

```
private Node parent;

public Optional<Node> getParent() {return Optional.ofNullable(parent);}
```

or initialize the parent field, e.g. 

```
private Optional<Node> parent = Optional.empty();
```